lib/mage/MageError.ts

```diff
@@ -0,0 +1,53 @@
+/**
+ * MageError details
+ */
+declare interface IErrorDetails {
+    /**
+     * The log level to use when logging this error (default: error)
+     *
+     * Some types of errors are perfectly normal; for instance,
+     * a user failing to authenticate may throw an error, but
+     * you might not want to have it logged as an error.
+     *
+     * Setting the level will allow you to decide whether this
+     * error should be ignored, considered benign, or serious.
+     */
+    level?: 'debug' | 'warning' | 'error' | 'alert' | 'emergency',
```

Comment:
Does debug have a place in this? Also, I think `critical` is missing.

Reply:
Who knows :D I would simply support all log levels just to make sure we have more choice.

Reply:
I take that back. The idea behind debug being in the list was that some userland errors are perfectly natural, and you wouldn't necessarily want to log them in production but still make sure to return a specific error code (insufficient funds, etc)